src/sentry/scim/endpoints/teams.py

```diff
@@ -440,8 +440,10 @@ def patch(self, request: Request, organization, team):
                             # delete all the current team members
                             # and replace with the ones in the operation list
                             with transaction.atomic(router.db_for_write(OrganizationMember)):
-                                queryset = OrganizationMemberTeam.objects.filter(team_id=team.id)
-                                queryset.delete()
+                                existing = list(
```

Comment:
Any scenarios where we should be worried about the size of this list now that we're materializing the records here first before deletion?

Reply:
Yes, and honestly that's the trick.  But fundamentally this existing design already has a huge problem with performance for our largest orgs.  That said, even querying 1000s of objects into materialize existence along an indexed column isn't a big issue so long as the underlying operation is "bulk" (that is, not an N+1 network interaction).  I believe this will still succeed well belong 5s even if it's a slow operation.

Long term, I think a more snappy implementation for our high end customers would be asynchronous (possibly even just using scheduled deletions).  I didn't feel comfortable making that change however as it would constitute an API semantic change.